app/services/os_version_updater.rb

```diff
@@ -0,0 +1,12 @@
+# frozen_string_literal: true
+
+# A service class to update OS version for each policy with assigned hosts
+class OsVersionUpdater
+  class << self
+    def run!
+      Policy.all.each do |policy|
+        policy.send(:update_os_versions, policy.hosts.map(&:id))
```

Comment:
IMO `update_os_versions` should just use the currently assigned hosts instead of a passed in list of IDs. That would make it fool-proof to use and always guarantee the correct result, and then it could be public without any issue so we don't have to `send` it.

Reply:
I don't know why I left it as `private` :thinking: sorry :see_no_evil: 

Regarding the list of IDs, I just wanted to avoid that +1 query for the `hosts.pluck(:id)` but I assume you are okay with it.

Reply:
We should be able to avoid N+1 by including hosts like `Policy.includes(:hosts).find_each`...